### PR TITLE
GENESIS-166 Merge translations only with existing languages, use en as fallback

### PR DESCRIPTION
**Pull request by jacekWojcikOpusCapita**
Adjust the Menu item language handling.

## Current behavior

When the user.userProfile -> languageId attribute includes a non supported value, the MainMenu breaks and it renders a blank page.

## Expected behavior

When there user's profile includes a non-supported languageId, ie. norwegian("no") or danish("da") , it should load the Menu items in English.

image

For the changes to take effect. The BNP service has to be updated. This might also affect other services that implements the menu from this library.

## Link

Jira [GENESIS-166](https://opuscapita.atlassian.net/browse/GENESIS-166)

### CODE REVIEW

File: components/MainMenu/MainMenu.react.js [hunk 1]
```diff
@@ -190,14 +190,16 @@ class MainMenu extends ConditionalRenderComponent
...
+        const { roles } = userData;
 
+        const userLocaleItems = navItems.businessPartner[locale] || [];
+        const userItems = this.recursiveMergeNavItems(navItems.businessPartner['en'], userLocaleItems);
+        let adminItems = []
         if(roles && roles.indexOf('admin') > -1)
-            items = this.recursiveMergeNavItems(items, navItems.admin[locale] || navItems.admin['en'])
+            adminItems = navItems.admin[locale] || navItems.admin['en'];
 
+        const items = this.recursiveMergeNavItems(userItems, adminItems);
```
Comment (nczeranka) [inline components/MainMenu/MainMenu.react.js:202]: I would improve method recursiveMergeNavItem by checking on null or an empty Array as parameters.
Taken this into acount the calling of the method could be simplified a little bit.

**Review by nczeranka — APPROVED** [review]
Okay for me.

File: components/MainMenu/MainMenu.react.js (hunk 1 above)
Comment (jaconsta) [reply components/MainMenu/MainMenu.react.js]: _recursiveMergeNavItems_ is adjusted for receiving null or empty arrays. I left _adminItems_ and _userItems_ for code readability.

**Comment by jacekWojcikOpusCapita** [comment]
Restored branch. The current code does not solve the problem described in the ticket.

**Pull request merged**